src/plugins/unified_field_list/public/components/field_stats/field_stats.tsx

```diff
@@ -210,10 +213,12 @@ const FieldStatsComponent: React.FC<FieldStatsProps> = ({
     fetchData();
 
     return () => {
-      isCanceledRef.current = true;
+      prevFetchedRef.current = fieldId;
       abortControllerRef.current?.abort();
     };
-  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+    // Need to refetch data when the field or query changes
+    // not just when popover is closed/component is unmounted
+  }, [fieldId, dslQuery]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I think changes to `dslQuery` would be ignored, because there is now the following condition in the fetch request:
```
if (prevFetchedRef.current === fieldId) {
  return;
}
```

Reply:
Updated here [`2ab4d99` (#147322)](https://github.com/elastic/kibana/pull/147322/commits/2ab4d991e26d302b6865b78967aef8d060957c48)